lib/fluent/plugin/kubernetes_metadata_cache_strategy.rb

```diff
@@ -24,12 +24,12 @@ def get_pod_metadata(key, namespace_name, pod_name, record_create_time, batch_mi
       if !ids.nil?
         # FAST PATH
         # Cache hit, fetch metadata from the cache
-        metadata = @cache.fetch(ids[:pod_id]) do
+        metadata = @cache.fetch(ids['pod_id']) do
           @stats.bump(:pod_cache_miss)
           m = fetch_pod_metadata(namespace_name, pod_name)
           (m.nil? || m.empty?) ? {'pod_id'=>ids[:pod_id]} : m
         end
-        metadata.merge!(@namespace_cache.fetch(ids[:namespace_id]) do
+        metadata.merge!(@namespace_cache.fetch(ids['namespace_id']) do
```

Comment:
why are we using a string and not the symbol?

Reply:
Rolled back the change.